distributed/shuffle/_merge.py

```diff
@@ -361,18 +361,16 @@ def cull(self, keys: Iterable[str], all_keys: Any) -> tuple[HashJoinP2PLayer, di
 
     def _construct_graph(self) -> dict[tuple | str, tuple]:
         token_left = tokenize(
-            "hash-join",
+            self.name,
```

Comment:
```suggestion
            # Generate unique shuffle IDs if the input frame is the same but
            # parameters differ. Reusing shuffles in merges is dangerous because of the
            # required coordination and complexity introduced through dynamic clusters.
            self.name,
```